code/model/Learner.cpp

```diff
@@ -109,6 +115,15 @@ bool SubgoalLearner::Init (void)
 	return (-1 != i_DomainPddlId);
 }
 
+//										
+void SubgoalLearner::OnIRAnswer (int _iIndex, IRAnswer& _aAnswer)
+{
+	// pthread_mutex_lock (&mtx_WaitForSequences);
+	// TODO: do something when IRAnswer is received
+	// pthread_mutex_unlock (&mtx_WaitForSequences);
+}
+
+
```

Comment:
Hmm,

I was thinking Policy constructs subgoals / plans and Learner uses that to learn given reward failure and interfaces with Solver

i thought answering questions fell under policy as a result

idk, it's your call. Whatever feels most elegant / fastest